File: mailblastr-cli/src/commands/logs.js
'use strict';

const { clean, toInt, withPagination, pagination } = require('../helpers');

function register({ group, leaf, act }) {
  const logs = group('logs', 'Inspect API request logs');

  act(
    withPagination(
      leaf(logs, 'list', 'List API request logs')
        .option('--method <method>', "filter by HTTP method, e.g. 'POST'")
        .option('--status <status>', 'filter by response status, e.g. 429'),
    ),
    ({ client, opts }) =>
      client.logs.list(
        clean({ ...pagination(opts), method: opts.method, status: toInt(opts.status, '--status') }),
      ),
  );

  act(leaf(logs, 'get <id>', 'Retrieve a log entry'), ({ client, args: [id] }) => client.logs.get(id));
}

module.exports = { register };

File: mailblastr-ruby/lib/mailblastr/logs.rb
# frozen_string_literal: true

module Mailblastr
  # API request logs.
  module Logs
    class << self
      # List logs — cursor pagination plus optional server-side `method`
      # (e.g. "POST") and `status` (e.g. 429) filters. GET /logs
      def list(params = {})
        query = Client.pagination(params)
        method = Client.opt(params, :method)
        status = Client.opt(params, :status)
        query[:method] = method if method
        query[:status] = status unless status.nil?
        Client.request(:get, "/logs", query: query)
      end

      # Retrieve one log with request/response bodies. GET /logs/:id
      def get(log_id)
        Client.request(:get, "/logs/#{Client.path_escape(log_id)}")
      end
    end
  end
end
